st2actions/st2actions/handlers/mistral.py

```diff
@@ -62,22 +62,32 @@ def callback(url, context, status, result):
                 # tracker before sending the output.
                 result.pop('tasks', None)
 
+            match = re.search('(.*)/action_executions/(.*)', url)
+            if not match or len(match.groups()) != 2:
+                raise Exception('Unable to extract the action execution ID '
+                                'from the callback URL (%s).' % url)
+
+            action_execution_id = match.group(2)
             output = json.dumps(result) if type(result) in [dict, list] else str(result)
             data = {'state': STATUS_MAP[status], 'output': output}
 
+            client = mistral.client(
+                mistral_url=cfg.CONF.mistral.v2_base_url,
+                username=cfg.CONF.mistral.keystone_username,
+                api_key=cfg.CONF.mistral.keystone_password,
+                project_name=cfg.CONF.mistral.keystone_project_name,
+                auth_url=cfg.CONF.mistral.keystone_auth_url)
+
+            manager = action_executions.ActionExecutionManager(client)
+
             for i in range(cfg.CONF.mistral.max_attempts):
                 try:
                     LOG.info('Sending callback to %s with data %s.', url, data)
-                    response = requests.request(method, url, data=json.dumps(data), headers=headers)
-                    if response.status_code == 200:
-                        break
+                    manager.update(action_execution_id, **data)
+                    break
                 except requests.exceptions.ConnectionError as conn_exc:
```

Comment:
Does the Mistral Python client throw the same exception if there is a connection issue (just want to confirm that)?